fubectl.source

```diff
@@ -1,30 +1,54 @@
 # helper functions
 alias _inline_fzf="fzf --multi --ansi -i -1 --height=50% --reverse -0 --header-lines=1 --inline-info --border"
 alias _inline_fzf_nh="fzf --multi --ansi -i -1 --height=50% --reverse -0 --inline-info --border"
+_isNamespaced(){
+    if [[ "$1" == "" ]]; then
+        echo "please specify a Kubernetes object!"
+        exit 1
+    fi
+    # ask cluster about not namespaced resources
+    if [ $(kubectl api-resources --no-headers --namespaced=false | awk '{print $1,$2,$4}' | sed 's/ /\n/g' | grep --count -i "^$1$")  -gt 0 ]; then
+        return 0
+   else
+        return 1
+    fi
+}
 
 # [k] like g for git but 233% as effective!
 alias k="kubectl"
 
-# [kall] get all pods
+# [ka] get all pods in namespace
+alias ka="kubectl get pods"
+
+# [kall] get all pods in cluster
 alias kall="kubectl get pods --all-namespaces"
 
-# [kwall] watch all pods
+# [kwa] watch all pods in namespace
+alias kwa="watch kubectl get pods"
+
+# [kwall] watch all pods in cluster
 alias kwall="watch kubectl get pods --all-namespaces"
 
+# TODO check if this works for macs
 # [kp] open kubernetes dashboard with proxy
-alias kp="open 'http://localhost:8001/ui' && kubectl proxy"
+alias kp="xdg-open 'http://localhost:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/' && kubectl proxy"
```

Comment:
```
foo && bar
```
This code only starts `bar` after `foo` returns from execution. `xdg-open` does not (necessarily) return execution. It can continue to run in foreground until you close your browser. In fact on the first call it blocks and in following calls (when a browser is open already) it will return control-flow. The `kp` alias will fail to start the proxy in all first uses.

If you need the proxy to run for the browser you start, then you should really start the proxy first and then open the browser. If you do not want to do that consider using the `&` instead of `&&` in your above example.
```
foo & bar
```
will start foo in the background and immediately run bar after that. (Still the wrong order, but it would at least work)

Reply:
I think `xdg-open` is starting in background anyway, so not blocking. I think the basic idea here is that the `kubectl proxy` execution is blocked, to be able to stop it with `^c`.

Reply:
You think `xdg-open` is starting in the background anyways, but I know that it does not necessarily do that.
On my machine `xdg-open` on a URL is blocking when there is no browser running, because it will start a browser and attach the tty to the calling terminal. On future `xdg-open` calls it will detach and run in the background (because it only sends the url to the running browser).

Replacing the `&&` with the `&` is the easiest way forward because it will make sure to run the xdg in the background.